### ISSUE TITLE
Unable to use custom container images

### ISSUE THREAD

**Issue by nmaludy** [puppetlabs/puppet_litmus#242]
## Describe the Bug
When trying to utilize a custom container image we receive an error.

## Expected Behavior
We're able to use our custom container images.

## Steps to Reproduce
Setup your `provision.yaml` like so:
```yaml
---
docker_centos7:
  provisioner: docker
  images: ['stackstorm/packagingtest:centos7-systemd']
```

Try to provision
```
$ bundle exec rake 'litmus:provision_list[docker_centos7]'
Provisioning stackstorm/packagingtest:centos7-systemd using docker provisioner.[✔]
rake aborted!
Failed to provision with 'docker'
 =====
localhost

[{"target"=>"localhost", "action"=>"task", "object"=>"provision::docker", "status"=>"failure", "result"=>{"_error"=>{"kind"=>"provision/docker_failure", "msg"=>"platform stackstorm_packagingtest not yet supported on docker", "backtrace"=>["/tmp/33d617ec-46b7-4ce2-bbfb-189f9bdc7b26/provision/tasks/docker.rb:47:in `install_ssh_components'", "/tmp/33d617ec-46b7-4ce2-bbfb-189f9bdc7b26/provision/tasks/docker.rb:99:in `provision'", "/tmp/33d617ec-46b7-4ce2-bbfb-189f9bdc7b26/provision/tasks/docker.rb:150:in `<main>'"]}}, "node"=>"localhost"}]
/home/nmaludy/storage/home/src/git/puppet-st2/vendor/bundle/ruby/2.5.0/gems/puppet_litmus-0.15.0/lib/puppet_litmus/rake_tasks.rb:63:in `block (2 levels) in <top (required)>'
/home/nmaludy/storage/home/src/git/puppet-st2/vendor/bundle/ruby/2.5.0/gems/rake-12.3.3/exe/rake:27:in `<top (required)>'
```

## Environment
 - Version: 0.15.0
 - Platform: Centos 7

## Addition Information
It looks like this is caused by some regex patching going on in the `provision` module: https://github.com/puppetlabs/provision/blob/master/tasks/docker.rb#L81
https://github.com/puppetlabs/provision/blob/master/tasks/docker.rb#L7-L53

Basically if you don't conform to the regex where you have the OS "type" in the docker container name (in our case have it in a tag) then this error is thrown.

**Comment by jordanbreen28** [comment]
Hey @nmaludy, thanks for raising this.
I'm auditing the ageing issues on this repo, and as I can't seem to replicate the issue with the info provided, I am going to set this issue as resolved.

```
bundle exec rake 'litmus:provision_list[docker_centos7]'
[✔] Provisioning stackstorm/packagingtest:centos7-systemd using docker provisioner.
localhost:52720, stackstorm/packagingtest:centos7-systemd
```

```
  - uri: localhost:52720
    config:
      transport: ssh
      ssh:
        user: root
        password: root
        port: 52720
        host-key-check: false
    facts:
      provisioner: docker
      container_name: stackstorm_packagingtest_centos7-systemd-52720
      platform: stackstorm/packagingtest:centos7-systemd
      os-release:
        NAME: CentOS Linux
        VERSION: 7 (Core)
        ID: centos
        ID_LIKE: rhel fedora
        VERSION_ID: '7'
        PRETTY_NAME: CentOS Linux 7 (Core)
        ANSI_COLOR: 0;31
        CPE_NAME: cpe:/o:centos:centos:7
        HOME_URL: https://www.centos.org/
        BUG_REPORT_URL: https://bugs.centos.org/
        CENTOS_MANTISBT_PROJECT: CentOS-7
        CENTOS_MANTISBT_PROJECT_VERSION: '7'
        REDHAT_SUPPORT_PRODUCT: centos
        REDHAT_SUPPORT_PRODUCT_VERSION: '7'
```

Please re-open if you find this issue still persists.
